server/bootstrap.ts

```diff
@@ -0,0 +1,44 @@
+import { Action, Event } from '@strapi/database/lib/lifecycles';
+import { Strapi } from '@strapi/strapi';
+
+export default ({ strapi }: { strapi: Strapi }) => {
+  const generateBlurhash = async (event: Event, eventType: Action) => {
+    const { data, where } = event.params;
+
+    if (data.mime && data.mime.startsWith('image/')) {
+      data.blurhash = await strapi
+        .plugin('strapi-blurhash')
+        .service('blurhash')
+        .generateBlurhash(data.url);
+    }
+
+    if (
+      eventType === 'beforeUpdate' &&
+      strapi.plugin('strapi-blurhash').config('regenerateOnUpdate') === true
+    ) {
+      const fullData = await strapi.db.query('plugin::upload.file').findOne({
+        select: ['url', 'blurhash', 'name', 'mime'],
+        where,
+      });
+
+      if (
+        fullData.mime &&
+        fullData.mime.startsWith('image/') &&
+        !fullData.blurhash
+      ) {
+        data.blurhash = await strapi
+          .plugin('strapi-blurhash')
+          .service('blurhash')
+          .generateBlurhash(fullData.url);
+      }
+    }
+  };
+
+  strapi.db.lifecycles.subscribe({
+    // TODO: Remove this @ts-ignore
+    // @ts-ignore
+    models: ['plugin::upload.file'],
```

Comment:
This line gives the error:
```log
Argument of type '{
  models: string[];
  beforeCreate: (event: Event) => Promise<void>;
  beforeUpdate: (event: Event) => Promise<void>;
}'
is not assignable to parameter of type 'Subscriber'.
  Object literal may only specify known properties, and 'models' does not exist in type 'Subscriber'.ts(2345)
```

It's because the type definition of `strapi.db.lifecycles.subscribe` doesn't have any `models` property, although in the Strapi documentation it's used...

I'm not familiar with Strapi plugin development, as a consequence I can't figure out a solution without proper research. So I just ignored this because it should be working, and it's just a type issue.